site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+
+Normally, a separate `Container` is created for each widget that needs to have `properties` injected. In the demo application, we have two widgets that rely on application state - `WorkerContainer` and `WorkerForm`. Let's start with the `WorkerContainer`. `Containers` are given the same name as the widget that they contain with the suffix `Container` appended. To keep things organized, they are also stored in a different sub-package - `containers`.
+
+{% instruction 'Add the following imports to the `WorkerContainerContainer` in the `containers` sub-package' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:1-4 %}
+
+The first `import` gives the module access to the `Container` factory function which will be used to construct the container. The other `import` allow the module to use the `ApplicationContext` to extract state, use the `WorkerContainerProperties` to receive properties from its parent, and the `WorkerContainer` class that will be wrapped by the container.
+
+Next, we need to address the fact that the container has two places to get properties from - its parent widget and the `ApplicationContext`. To tell the container how to manage this, we will create a function called `getProperties`.
+
+{% instruction 'Add the `getProperties` function to the `WorkerContainerContainer` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:6-8 %}
+
+{% aside 'Additional injection options' %}
+`Injectors` are capable of injecting more than just properties into a widget. A `getChildren` mapping function can also be provided to the `Container`, allowing children to be injected into the widget as well. Just like the `getProperties` function, the `getChildren` function will receive the children from the parent widget and the injector. It then returns the final array of children that will be provided to the contained widget.
+{% endaside %}
+
+The `getProperties` function receives two parameters. The first is the `payload` of the `injector` instance, for the demo application this is the data passed when creating the `ApplicationContext`. The second is the `properties` that have been passed to the container via the normal mechanism, `w(Container, properties)`, the properties will implement the properties interface defined by the wrapped widget (for example `WorkerContainerProperties`). The `getProperties` function must then return an object that holds the properties that will be passed to the widget itself. In this example, we are ignoring the properties provided by the parent and returning the `workerData` stored by the `ApplicationContext`. More advanced use cases where both sources are used to generate the properties are also possible.
```

Comment:
", the properties will implement the properties interface" -> ". The properties will implement the properties interface"

Reply:
". The `getProperties` function" I would suggest this be the start of a new paragraph